core/src/main/java/io/lionweb/utils/ChunkValidator.java

```diff
@@ -0,0 +1,115 @@
+package io.lionweb.utils;
+
+import io.lionweb.serialization.data.*;
+import java.util.*;
+import javax.annotation.Nonnull;
+
+/**
+ * Validate a generic chunk, which may contain part of a partition, an entire partition, or perhaps
+ * multiple partitions.
+ */
+public class ChunkValidator extends Validator<SerializedChunk> {
```

Comment:
It'd be interesting (/Nice-To-Have) to see if you could make this “streaming” in the sense of processing 1 node at a time, and building up a data structure that's as small as possible but sufficient to do validation.